openstack/identity/v3/extensions/ec2tokens/requests.go

```diff
@@ -280,6 +299,29 @@ func Create(c *gophercloud.ServiceClient, opts tokens.AuthOptionsBuilder) (r tok
 	return
 }
 
+// ValidateS3Token authenticates an S3 request using EC2 credentials. Doesn't
+// generate a new token ID, but returns a tokens.CreateResult.
+func ValidateS3Token(c *gophercloud.ServiceClient, opts tokens.AuthOptionsBuilder) (r tokens.CreateResult) {
```

Comment:
How about doing something similar to `tokens.Validate` where `(bool, error)` is returned: https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/tokens/requests.go#L145

Reply:
What if we need to fetch user/project/domain info from the token response?

Reply:
Are you sure that's included? The debug output doesn't show it.

Reply:
It returns the same token body as usual, when keystone auths a user. The only difference is it doesn't return a token.

Reply:
see https://github.com/openstack/keystone/blob/40b7de87e04fbc49ab3a96b5114a9ab09c2734b7/keystone/api/s3tokens.py#L100..L105 vs https://github.com/openstack/keystone/blob/40b7de87e04fbc49ab3a96b5114a9ab09c2734b7/keystone/api/ec2tokens.py#L67..L73

Reply:
Oh, right - the output is being redacted/removed in the acceptance tests. OK, I don't want to bike shed this one. Perhaps the `Validate` function I linked to should be renamed to `IsValid` at some point in the future.

Reply:
Indeed debug doesn't show the json response, but content length is big and it should include the token data:

```
2020/03/27 13:58:13 [DEBUG] OpenStack Response Code: 200
2020/03/27 13:58:13 [DEBUG] OpenStack Response Headers: 
Content-Length: 6751
Content-Type: application/json
Date: Fri, 27 Mar 2020 13:58:13 GMT
Server: Apache/2.4.29 (Ubuntu)
Vary: X-Auth-Token
X-Openstack-Request-Id: req-07cf74fe-a252-4cbb-8107-0d263dc54b3e
```